From c979a97f81d2b5e81ff7a5ecbe41139dc25dcccb Mon Sep 17 00:00:00 2001
From: Cedric IZOARD <cedricx.izoard@intel.com>
Date: Mon, 14 Oct 2013 10:03:57 +0200
Subject: [PORT FROM MAIN]wifi: Fix KW issue

BZ: 144622

Fix KW issue introduced by patch :
WpaSupp: Don't keep hidden AP in the list after forget it

File: wpa_supplicant_8/wpa_supplicant/bss.c
Object 'bss' was dereferenced at line 739 after being freed by calling 'wpa_bss_remove' at line 737

Change-Id: Ic5f1a7543d2d22bfbf0f53f9f9db1caac1cf5f21
Orig-Change-Id: I5e11342d6a006b0d14d86453e5b76cd56f5addac
Category: device enablement
Domain: CWS.WIFI-Bluewoods
Origin: upstream
Upstream-Candidate: no,proprietary
Signed-off-by: Cedric IZOARD <cedricx.izoard@intel.com>
Reviewed-on: http://android.intel.com:8080/138191
Reviewed-by: Poulain, LoicX <loicx.poulain@intel.com>
Reviewed-by: Mauger, MatthieuX <matthieux.mauger@intel.com>
Reviewed-by: Samoun, JacquesX <jacquesx.samoun@intel.com>
Reviewed-by: Klouz, AymenX <aymenx.klouz@intel.com>
Reviewed-by: Trivelly, Jean <jean.trivelly@intel.com>
Reviewed-by: Garcia, Jeremie <jeremie.garcia@intel.com>
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 wpa_supplicant/bss.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/wpa_supplicant/bss.c b/wpa_supplicant/bss.c
index fa64fd7..83c3173 100644
--- a/wpa_supplicant/bss.c
+++ b/wpa_supplicant/bss.c
@@ -736,6 +736,7 @@ void wpa_bss_update_end(struct wpa_supplicant *wpa_s, struct scan_info *info,
 		if (bss->scan_miss_count >=
 		    wpa_s->conf->bss_expiration_scan_count) {
 			wpa_bss_remove(wpa_s, bss, "no match in scan");
+			continue;
 		}
 		if (bss->is_hidden) {
 			wpa_bss_remove(wpa_s, bss, "not connected hidden BSS");
-- 
1.7.4.1

